Pillar_II/esm/src/dummy_prune_analysis.py

```diff
@@ -7,37 +11,40 @@
 from pycompss.api.parameter import IN  # type: ignore
 from pycompss.api.task import task  # type: ignore
 from pycompss.runtime.management.classes import Future  # type: ignore
-from hecuba import StorageDict
-import time
-import random
-
-class MetaDictClass(StorageDict):
-   '''
-   @TypeSpec dict <<keyname0:str>,valuename0:str>
-   '''
-
-def simulation_started(expid):
-    state = MetaDictClass.get_by_alias(expid) 
-    if len([k for k in s.keys()]) > 1:
-        return True
-    else:
-        return False
+
+"""eFlows4HPC ensemble members pruning."""
+
+CHECK_FOR_PRUNING_SLEEP_TIME_SECS = 5
+"""How long should we sleep for"""
+
+logger = logging.getLogger(__name__)
+
 
 @task(expid=IN)
-def esm_analysis_prune(expid):
-    
+def esm_analysis_prune(expid: str):
+    """This has to be called by PYCOMPSs as analysis.
+
+    Args:
+        expid: The experiment ID.
+    """
+    # N.B.: importing this results in a network query to Hecuba servers,
+    #       which fails if the servers are not available.
+    from hecuba import StorageDict  # type: ignore
+    # This is an infinite-loop, with a sleep time. The execution
+    # must be wrapped in an existing COMPSs or Slurm job, with a
+    # walltime or some limit to control the maximum execution
+    # time, and kill this task.
     while True:
-        started = simulation_started(expid)
-        if started is True:
+        mdc = StorageDict.get_by_alias(expid)
+        if mdc:
```

Comment:
No need to import `MetaDictClass` here, we can call Hecuba's `StorageDict` directly (I had a look at this class, and found nothing that would require extending it, like Python meta classes, or at least not in the StorageDict class..). When this returns a valid object in Python we can check it's not `None` instead of relying on boolean values :+1: